src/example/EASPortal.sol

```diff
@@ -56,26 +62,30 @@ contract EASPortal is AbstractPortal {
       abi.encodePacked(attestationRequest.data.recipient),
       attestationRequest.data.data
     );
-
     super.attest(attestationPayload, validationPayload);
-  }
-
-  function bulkAttest(AttestationRequest[] memory attestationsRequests) public payable {
-    AttestationPayload[] memory attestationsPayloads = new AttestationPayload[](attestationsRequests.length);
-    bytes[][] memory validationPayloads = new bytes[][](attestationsRequests.length);
-
-    for (uint256 i = 0; i < attestationsRequests.length; i++) {
-      attestationsPayloads[i] = AttestationPayload(
-        attestationsRequests[i].schema,
-        attestationsRequests[i].data.expirationTime,
-        abi.encodePacked(attestationsRequests[i].data.recipient),
-        attestationsRequests[i].data.data
+    // if refUID exists then create relationship attestation
+    if (attestationRequest.data.refUID != 0) {
+      if (!attestationRegistry.isRegistered(attestationRequest.data.refUID)) revert ReferenceAttestationNotRegistered();
+
+      uint32 attestationIdCounter = attestationRegistry.getAttestationIdCounter();
+      bytes32 attestationId = bytes32(abi.encode(attestationIdCounter));
+
+      AttestationPayload memory relationshipAttestationPayload = AttestationPayload(
+        relationshipSchemaId,
+        attestationRequest.data.expirationTime,
+        abi.encodePacked(attestationRequest.data.refUID),
+        abi.encode(attestationId, "EASrefUID", attestationRequest.data.refUID)
       );
 
-      validationPayloads[i] = new bytes[](0);
+      super.attest(relationshipAttestationPayload, validationPayload);
     }
+  }
 
-    super.bulkAttest(attestationsPayloads, validationPayloads);
+  function bulkAttest(AttestationRequest[] memory attestationsRequests) external payable {
+    for (uint256 i = 0; i < attestationsRequests.length; i++) {
+      attest(attestationsRequests[i]);
+    }
+    emit BulkAttestationsRegistered();
```

Comment:
No need to emit this event, we've removed it from `Attestationregistry` too (to save some gas)